homeassistant/components/light/insteon_plm.py

```diff
@@ -0,0 +1,121 @@
+        """New device detected from transport."""
+        name = device['address']
+        address = device['address_hex']
+
+        dimmable = bool('dimmable' in device['capabilities'])
+
+        _LOGGER.info('New INSTEON PLM light device: %s (%s)', name, address)
+        hass.async_add_job(async_add_devices(
+            [InsteonPLMDimmerDevice(hass, plm, address, name, dimmable)]))
+
+    criteria = dict(capability='light')
+    plm.protocol.devices.add_device_callback(
+        async_plm_light_callback, criteria)
+
+    new_lights = []
+    yield from async_add_devices(new_lights)
+
+
+class InsteonPLMDimmerDevice(Light):
+    """A Class for an Insteon device."""
+
+    def __init__(self, hass, plm, address, name, dimmable):
+        """Initialize the light."""
+        self._hass = hass
+        self._plm = plm.protocol
+        self._address = address
+        self._name = name
+        self._dimmable = dimmable
+
+        self._plm.add_update_callback(
+            self.async_light_update, dict(address=self._address))
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def address(self):
+        """Return the the address of the node."""
+        return self._address
+
+    @property
+    def name(self):
+        """Return the the name of the node."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        onlevel = self._plm.get_device_attr(self._address, 'onlevel')
+        _LOGGER.debug('on level for %s is %s', self._address, onlevel)
+        return int(onlevel)
+
+    @property
+    def is_on(self):
+        """Return the boolean response if the node is on."""
+        onlevel = self._plm.get_device_attr(self._address, 'onlevel')
+        _LOGGER.debug('on level for %s is %s', self._address, onlevel)
+        return bool(onlevel)
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._dimmable:
+            return SUPPORT_BRIGHTNESS
+
+    @property
+    def device_state_attributes(self):
+        """Provide attributes for display on device card."""
+        insteon_plm = get_component('insteon_plm')
+        return insteon_plm.common_attributes(self)
+
+    def get_attr(self, key):
+        """Return specified attribute for this device."""
+        return self._plm.get_device_attr(self.address, key)
+
+    def async_light_update(self, message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.info('Received update calback from PLM for %s', self._address)
+        self._hass.async_add_job(self.async_update_ha_state(True))
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+        else:
+            brightness = 255
```

Comment:
Totally sensible, but any chance you'd consider making this a module level constant? [Magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants) are always a PITA even though your code is super straightforward.